diff -urN source2/Connection.py source/Connection.py
--- source2/Connection.py	2007-04-17 01:39:34.000000000 +0000
+++ source/Connection.py	2007-05-29 20:24:06.881131227 +0000
@@ -48,9 +48,9 @@
 	    #LOGGING IN FROM COOKIE
 	    cookie = ""
 	    cookie_details = True
-	    thereis_cookie = os.path.exists(os.path.join(globals.sourcefolder,globals.cookiefile))
+	    thereis_cookie = os.path.exists(os.path.join(globals.confdir,globals.cookiefile))
 	    if usecookie and thereis_cookie:
-		    cookie_details = pickle.load(file(os.path.join(globals.sourcefolder,globals.cookiefile),"rb"))
+		    cookie_details = pickle.load(file(os.path.join(globals.confdir,globals.cookiefile),"rb"))
 		    time_17min = 1020
 
 		    if time.time() < cookie_details["time"] + time_17min:
@@ -89,14 +89,14 @@
 			    globals.disable_osdb = False
 			    message = "Succesful Connection from XMLRPC, username = %s" % globals.preferences_list["username"]
 			    cookie = {"token": login["token"],"time": time.time(),"username": globals.logged_as}
-			    pickle.dump(cookie,file(os.path.join(globals.sourcefolder,globals.cookiefile),"wb"))
+			    pickle.dump(cookie,file(os.path.join(globals.confdir,globals.cookiefile),"wb"))
 			    return True,message
 		else:
 		    message = "Anonymous access withouth XMLRPC"
 		    globals.logged_as = ""
 		    globals.disable_osdb = False
 		    cookie = {"token":"00000","time": time.time(),"username": ""}
-		    pickle.dump(cookie,file(os.path.join(globals.sourcefolder,globals.cookiefile),"wb"))
+		    pickle.dump(cookie,file(os.path.join(globals.confdir,globals.cookiefile),"wb"))
 		    return True,message
 			    
 	    except:
@@ -135,7 +135,7 @@
 			    globals.sublanguages["languages_name2xxx"] = languages_name2xxx
 			    globals.sublanguages["languages_xx2xxx"] = languages_xx2xxx
 			    globals.sublanguages["total_subtitles_languages"] = int(server_details["total_subtitles_languages"])
-			    pickle.dump(globals.sublanguages,file(os.path.join(globals.sourcefolder,"conf/.sublanguages"),"wb"))
+			    pickle.dump(globals.sublanguages,file(os.path.join(globals.confdir,".sublanguages"),"wb"))
 	    except:
 		    globals.Log("Error globals.xmlrpc_server.GetSubLanguages")
     
\ No newline at end of file
diff -urN source2/PanelDownload.py source/PanelDownload.py
--- source2/PanelDownload.py	2007-04-16 15:45:02.000000000 +0000
+++ source/PanelDownload.py	2007-05-29 20:29:13.893364309 +0000
@@ -228,7 +228,7 @@
 	video_filename = os.path.join(video_data["dirname"],video_data["filename"])
 	subfile_id = sub_data["IDSubtitleFile"]	
 	
-	path_sub_temp = os.path.join(globals.sourcefolder,"conf","temp.srt")
+	path_sub_temp = os.path.join(globals.confdir,"temp.srt")
 	self.download_dlg = PP.PyProgress(None, -1, _("Downloading"),
                             _("Downloading, it can take a while..."),                            
                             style = wx.PD_CAN_ABORT)
@@ -263,7 +263,7 @@
 	video_filename = os.path.join(video_data["dirname"],video_data["filename"])
 	subfile_id = sub_data["IDSubtitleFile"]	
 	
-	path_sub_temp = os.path.join(globals.sourcefolder,"conf","temp.srt")
+	path_sub_temp = os.path.join(globals.confdir,"temp.srt")
 	self.download_dlg = PP.PyProgress(None, -1, _("Downloading"),
                             _("Downloading, it can take a while..."),                            
                             style = wx.PD_CAN_ABORT)
diff -urN source2/SubDownloader.py source/SubDownloader.py
--- source2/SubDownloader.py	2007-04-17 00:37:30.000000000 +0000
+++ source/SubDownloader.py	2007-05-29 21:13:14.091909474 +0000
@@ -41,7 +41,7 @@
 			#Default preferences
 		
 		try:
-			globals.sublanguages =  pickle.load(file(os.path.join(globals.sourcefolder,"conf/.sublanguages"),"rb"))		
+			globals.sublanguages =  pickle.load(file(os.path.join(globals.sourcefolder,"conf",".sublanguages"),"rb"))		
 		except:
 			globals.sublanguages = {}
 			globals.sublanguages["total_subtitles_languages"] = 0
@@ -64,6 +64,7 @@
 		
 		self.DefaultValue("search_langs","all")
 		self.DefaultValue("show_tips",True)
+		#self.DefaultValue("show_choose_languages",False)
 		self.DefaultValue("show_choose_languages",True)
 		
 		self.DefaultValue("server_osdb","http://www.opensubtitles.org/xml-rpc")
@@ -319,8 +320,8 @@
 		globals.param_function = status_before
 	       
 		if activated_gui:
-		    if os.path.exists(os.path.join(globals.sourcefolder,cachefile)):
-			f = open(os.path.join(globals.sourcefolder,cachefile),"aw")
+		    if os.path.exists(os.path.join(globals.confdir,cachefile)):
+			f = open(os.path.join(globals.confdir,cachefile),"aw")
 			f.write(param)
 			f.write("\n")
 			f.close()
@@ -328,7 +329,7 @@
 			    stop_running = True
 		    else:
 			#We create the file and launch the GUI
-			f = open(os.path.join(globals.sourcefolder,cachefile),"w")
+			f = open(os.path.join(globals.confdir,cachefile),"w")
 			f.write(param)
 			f.write("\n")
 			f.close()
@@ -345,27 +346,40 @@
 	    if stop_running:
 		sys.exit(0)
 	else:	    
-		if os.path.exists(os.path.join(globals.sourcefolder,".cachesearch")):
-		    os.remove(os.path.join(globals.sourcefolder,".cachesearch"))
+		if os.path.exists(os.path.join(globals.confdir,".cachesearch")):
+		    os.remove(os.path.join(globals.confdir,".cachesearch"))
 		
-		if os.path.exists(os.path.join(globals.sourcefolder,".cacheupload")):
-		    os.remove(os.path.join(globals.sourcefolder,".cacheupload"))
+		if os.path.exists(os.path.join(globals.confdir,".cacheupload")):
+		    os.remove(os.path.join(globals.confdir,".cacheupload"))
 		    
     def OnInit(self):
-	globals.sourcefolder = os.path.dirname(sys.argv[0])
+
+	globals.sourcefolder = os.path.dirname(sys.argv[0])
 	globals.subdownloader_pointer = self
 	self.log_memory = []
 	
 	if platform.system() == "Linux":
 	    home = os.getenv("HOME")
 	    if not os.path.isdir(os.path.join(home,".subdownloader")):
-		os.mkdir(os.path.join(home,".subdownloader"))
-	    globals.preferences_filename = os.path.join(home,".subdownloader","preferences.conf")
+		os.mkdir(os.path.join(home,".subdownloader"))
+		globals.first_run = True
+		globals.preferences_filename = os.path.join(globals.sourcefolder,"conf","preferences.conf")
+		globals.confdir = os.path.join(globals.sourcefolder,"conf")
+	    else:
+		globals.preferences_filename = os.path.join(home,".subdownloader","preferences.conf")
+		globals.confdir = os.path.join(home,".subdownloader")
 	else:
-	    globals.preferences_filename = os.path.join(globals.sourcefolder,"conf","preferences.conf")
-	    
+	    globals.preferences_filename = os.path.join(globals.sourcefolder,"conf","preferences.conf")
+	    globals.confdir = os.path.join(globals.sourcefolder,"conf")
+	    
 	self.ManageArguments()
 	self.LoadPreferences()
+	
+	if globals.first_run:
+	    globals.preferences_filename = os.path.join(home,".subdownloader","preferences.conf")
+	    globals.confdir = os.path.join(home,".subdownloader")
+	    globals.first_run = False
+	    
 	globals.Log("-------------Showing preferences:")
 	for key,value in globals.preferences_list.items():
 	    if key != "password":
diff -urN source2/globals.py source/globals.py
--- source2/globals.py	2007-04-17 00:38:02.000000000 +0000
+++ source/globals.py	2007-05-29 20:41:07.494320140 +0000
@@ -43,7 +43,9 @@
 imdbserver = imdb.IMDb()
 
 #GLOBAL NEEDED(don't change)
-cookiefile = "conf/.cookie"
+cookiefile = ".cookie"
+confdir = ""
+first_run = False
 sourcefolder = ""
 app_parameteres = ""
 
